src/org/exoplatform/controller/document/DocumentAdapter.java

```diff
@@ -67,123 +68,165 @@ public long getItemId(int pos) {
-                    convertView.setBackgroundResource(R.drawable.dashboard_single_background_shape);
-                else {
-                    convertView.setBackgroundResource(R.drawable.dashboard_top_background_shape);
-                }
-            } else {
-                if (position + 1 == _documentList.size())
-                    convertView.setBackgroundResource(R.drawable.dasboard_bottom_background_shape);
-                else {
-                    ExoFile previousItem = _documentList.get(position - 1);
-                    ExoFile nextItem = _documentList.get(position + 1);
-
-                    if ("".equals(previousItem.name) && "".equals(previousItem.path)
-                            && "".equals(nextItem.name) && "".equals(nextItem.path)) {
-                        convertView.setBackgroundResource(R.drawable.dashboard_single_background_shape);
-                    } else if ("".equals(previousItem.name) && "".equals(previousItem.path)) {
-                        convertView.setBackgroundResource(R.drawable.dashboard_top_background_shape);
-                    } else if ("".equals(nextItem.name) && "".equals(nextItem.path))
-                        convertView.setBackgroundResource(R.drawable.dasboard_bottom_background_shape);
-                    else
-                        convertView.setBackgroundResource(R.drawable.dashboard_middle_background_shape);
-                }
+  @Override
+  public int getViewTypeCount() {
+    // there are 2 type, for gadget tab and for file item tab;
+    return 2;
+  }
 
-            }
+  @Override
+  public int getItemViewType(int position) {
+    int ret = VIEW_TYPE_GADGET;
+    final ExoFile myFile = _documentList.get(position);
+    if ("".equals(myFile.name) && "".equals(myFile.path)) {
+      ret = VIEW_TYPE_GADGET;
+    } else {
+      ret = VIEW_TYPE_FILEITEM;
+    }
+    return ret;
+  }
+  
+  static class GadgetHolder {
+    TextView textViewTabTitle;
+  }
+  static class FileHolder {
+    Button btnAction;
+    ImageView icon;
+    TextView lb;
+  }
+  
+  @Override
+  public View getView(int position, View convertView, ViewGroup parent) {
+
+    final int pos = position;
+    LayoutInflater inflater = (LayoutInflater) _mContext.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
+    // View rowView = inflater.inflate(R.layout.fileitem, parent, false);
+
+    final ExoFile myFile = _documentList.get(pos);
+    int viewType = getItemViewType(position);
+    if (viewType == VIEW_TYPE_GADGET) {
+      GadgetHolder holder;
+      if (convertView == null) {
+        convertView = inflater.inflate(R.layout.gadget_tab_layout, parent, false);
+        holder = new GadgetHolder();
+        holder.textViewTabTitle = (TextView) convertView.findViewById(R.id.textView_Tab_Title);
+        convertView.setTag(holder);
+      } else {
+        holder = (GadgetHolder) convertView.getTag();
+      }
+      if (myFile.driveName.equals(ExoConstants.DOCUMENT_PERSONAL_DRIVER))
```

Comment:
Reverse the condition order `ExoConstants.DOCUMENT_PERSONAL_DRIVER.equals(myFile.driveName)`